bin/add_frontend_server.rb

```diff
@@ -241,10 +272,12 @@ def execute_ssh_on_channel(ssh, command, exit_error_string)
 cmd = "ssh gateway.code.org -t knife bootstrap #{private_dns_name} -x ubuntu --sudo -E #{environment} -N #{instance_name} -r role[#{role}]"
```

Comment:
It would be a lot nicer to have the input scroll through the terminal for this step, seeing as how it takes a long time.

This can be done using open_channel. You can print stdout and std_err with the on_data and on_extended_data handler

http://net-ssh.github.io/net-ssh/